### PR TITLE
Add base template for the Docker/k8s workshops

### PR DESCRIPTION
**Pull request by carlosafonso**
This PR adds the base CloudFormation template to be used in the Docker and Kubernetes workshops. The template provides a common working environment for all workshop users, which includes a VPC with a public subnet and an EC2 instance with Docker and Docker Compose already installed.

### CODE REVIEW

File: docker-working-env-base.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+AWSTemplateFormatVersion: 2010-09-09
+
+Description: This template deploys a VPC and an EC2 instance with Docker already installed, with the goal of standardizing the experience of the users for the Docker and k8s workshops.
+
+Parameters:
+  KeyPair:
+    Type: AWS::EC2::KeyPair::KeyName
+    Description: The EC2 key pair to be used for SSHing into the instance
+
+Mappings:
```
Comment (pjlewisuk) [inline docker-working-env-base.yaml]: I think we should use the SSM lookup method here to keep the template short and prevent the need for regular updates. You can see an example (for Amazon Linux 1) here: https://github.com/awslabs/amazon-ecs-nodejs-microservices/pull/8/commits/01e2615f7cff972c96b23921fb7b3a878dbf42f5

File: docker-working-env-base.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      GroupDescription: Security group for the Docker/k8s workshop EC2 instance
+      SecurityGroupIngress:
+        - CidrIp: 0.0.0.0/0
+          IpProtocol: -1
+      VpcId: !Ref VPC
+
+  Instance:
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId: !FindInMap [RegionToAmiId, !Ref "AWS::Region", AMI]
```
Comment (pjlewisuk) [inline docker-working-env-base.yaml]: This will need to change to reference the SSM AMI resource

**Review by pjlewisuk — CHANGES_REQUESTED** [review]
Hi Carlos. I think we should use the SSM-lookup method (https://aws.amazon.com/blogs/compute/query-for-the-latest-amazon-linux-ami-ids-using-aws-systems-manager-parameter-store/) instead of AMI mappings. I've linked to an example PR from another repo for you to reference.

**Comment by carlosafonso** [comment]
Got it! Working on it. Thanks.

**Comment by carlosafonso** [comment]
Done, using param from SSM: `/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2`.

**Review by pjlewisuk — APPROVED** [review]
LGTM

**Pull request merged**